### PR TITLE
Screen reader accessibility 

### PR DESCRIPTION
**Pull request by IeuanWalker**
Wanted to make a draft PR just so you guys know I'm looking into it.

The code changes I've done seam to be working fine, just want to do a few more tests

#15

### CODE REVIEW

**Comment by LuckyDucko** [comment]
@IeuanWalker cheers for the PR, really appreciate it. I'll give it a review hopefully tonight

**Comment by IeuanWalker** [comment]
> @IeuanWalker cheers for the PR, really appreciate it. I'll give it a review hopefully tonight

Thanks, i just got to test different scenarios, so I'd hold off reviewing for now. I wont have much time today to do anything.

**Comment by IeuanWalker** [comment]
@LuckyDucko this is ready for review now.

I've tested the following type of apps -
- Sample app
- Shell app
- Tabled page app
	- This bug doesn't seem to be an issue anymore, so the workaround wasn't needed (https://github.com/rotorgames/Rg.Plugins.Popup/issues/721)
- Flyout (MasterDetailPage) 

Here's a test solution with the different types of projects, if you want to give it a go - 
[Test.zip](https://github.com/LuckyDucko/Mopups/files/9104809/Test.zip)

**Comment by IeuanWalker** [comment]
@LuckyDucko Did notice iOS doesn't render the popups correctly, you can see this in the test project I attached above.

- The popup is cut off at the bottom
- Looks like the background is rendered twice with one having a margin added, so there is a darker inner background

image

Want me to create a separate issue for this?

**Comment by LuckyDucko** [comment]
@IeuanWalker You know what, I remember seeing that. But then I believe I got sidetracked concerning how the iOS keyboard dismissal services worked, so locked up in that I missed it wasn't rendering properly. Definitely need a new issue for that, and assign it to me if possible so I can get onto it. 

I'll review this tonight and get a crack at the iOS popup, I bet its something simple, but Visual Studio for Mac in preview is a bit...dicey right now

**Comment by IeuanWalker** [comment]
@LuckyDucko No prob, once this is merged I take a look at the keyboard navigation side of things

**Comment by IeuanWalker** [comment]
@LuckyDucko think I have a solution for the keyboard accessibility.

Once this is merged will look at doing a PR for it

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (LuckyDucko) [inline Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:98]: How often is it that the MainPage is null? Do you think this should be a silent return, or an exception thrown?

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (LuckyDucko) [inline Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:107]: Could we switch this to using ?. instead of the if statement

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (LuckyDucko) [inline Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:115]: Could we switch this to using ?. instead of the if statement

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (LuckyDucko) [inline Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:124]: Could we switch this to using ?. instead of the if statement

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (LuckyDucko) [inline Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:58]: Is the name "PlatformRenderer" something under our control? or MAUI's? Just getting a quick grip with how this works

**Review by LuckyDucko — CHANGES_REQUESTED** [review]

**Comment by LuckyDucko** [comment]
@IeuanWalker  i was about to mention A11yTools as potentially something that could help you out, but then i realised you had actually contributed to that repo, so right now i'm coming up empty for helpful side notes.

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (IeuanWalker) [reply Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:98]: None that I have come across. 

But checking for null at that point means the nav count variables don't become nullable `(int?)` meaning we don't have to cast them to an int when getting the page from the stack

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (IeuanWalker) [reply Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:115]: Same as previous comment

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (IeuanWalker) [reply Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:124]: Same as previous comment

File: Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs
Comment (LuckyDucko) [reply Mopups/Mopups.Maui/Platforms/Android/Impl/AndroidMopups.cs:98]: Ah excellent point 👍🏻 just am wary of early returns. All good

**Comment by LuckyDucko** [comment]
Cheers for this @IeuanWalker amazing work and an excellent PR. The question about platform renderer is more just a question, so don't worry too much about answering it, i'll stumble across it eventually ahaha

**Comment by IeuanWalker** [comment]
@LuckyDucko Thanks.

I don't think we actually need that function anymore. It was something I copied from Pure Wenn A11y repo, to fix an issue in xamarin. Just tested it with it commented out and it was fine. And `View` in that method always returned null, so wasn't actually doing anything.

https://github.com/LuckyDucko/Mopups/pull/16

Are you ok if I remove it, when I do the PR for the keyboard stuff?

**Comment by LuckyDucko** [comment]
@IeuanWalker if it appears to not do too much, i'll probably remove it when i do the release tomorrow

**Comment by IeuanWalker** [comment]
@LuckyDucko iv done a PR for the keyboard side of things, that also removes.

Could u look at that before the release?

**Pull request merged**
